src/stores/index.ts

```diff
@@ -0,0 +1,41 @@
+import { configureStore, createListenerMiddleware } from '@reduxjs/toolkit';
+import { CHAT_ID_STORE_KEY, CHAT_MESSAGES_STORE_KEY } from 'src/constants';
+import chatReducer from 'src/stores/chat-slice';
+
+export type RootState = {
+	chat: ReturnType< typeof chatReducer >;
+};
+
+const listenerMiddleware = createListenerMiddleware< RootState >();
+
+listenerMiddleware.startListening( {
+	predicate( action, currentState, previousState ) {
+		return currentState.chat.messagesDict !== previousState.chat.messagesDict;
+	},
+	effect( action, listenerApi ) {
+		const state = listenerApi.getState() as RootState;
```

Comment:
It's redundant, sicne you type middleware [here](https://github.com/Automattic/studio/pull/826/files#diff-a962836e7056c855240fcab71b4e4f6a56e8462e298844f0b928226d230cda83R9)
```suggestion
		const state = listenerApi.getState();
```